### ISSUE TITLE
Debug why "Another task is already in progress" when detaching/attaching disks quickly

### ISSUE THREAD

**Issue by fdawg4l** [vmware/vic#1725]
```
=== RUN   TestListImageStore
time="2016-07-28T18:33:38Z" level=info msg="Creating directory [datastore1] dd024095-be66-47-imageTests" 
time="2016-07-28T18:33:38Z" level=info msg="Datastore path is [datastore1] dd024095-be66-47-imageTests" 
time="2016-07-28T18:33:38Z" level=info msg="Refreshing image cache from datastore." 
time="2016-07-28T18:33:38Z" level=info msg="Creating directory [datastore1] dd024095-be66-47-imageTests/storeName0/images" 
time="2016-07-28T18:33:38Z" level=info msg="Creating directory [datastore1] dd024095-be66-47-imageTests/storeName0/images/scratch" 
time="2016-07-28T18:33:38Z" level=info msg="Creating image scratch ([datastore1] dd024095-be66-47-imageTests/storeName0/images/scratch/scratch.vmdk)" 
time="2016-07-28T18:33:38Z" level=info msg="Create/attach vmdk [datastore1] dd024095-be66-47-imageTests/storeName0/images/scratch/scratch.vmdk from parent " 
time="2016-07-28T18:33:38Z" level=error msg="vmdk storage driver failed to attach disk: Another task is already in progress." 
--- FAIL: TestListImageStore (1.14s)
        Error Trace:    image_test.go:129
        Error:          Received unexpected error "Another task is already in progress."
```

```
=== RUN   TestCreateFS
time="2016-07-28T18:33:36Z" level=debug msg="Trying to log in with username/password in lieu of cert" 
time="2016-07-28T18:33:36Z" level=debug msg="Session Environment Info: " API Type=HostAgent API Version=6.0 Build=3634798 Name="VMware ESXi" OS Type=vmnix-x86 Product ID=embeddedEsx UUID= Vendor="VMware, Inc." Version=6.0.0 
time="2016-07-28T18:33:36Z" level=debug msg="vSphere resource cache populating..." 
time="2016-07-28T18:33:36Z" level=debug msg="Cached dc: " 
time="2016-07-28T18:33:36Z" level=debug msg="Cached cluster: " 
time="2016-07-28T18:33:36Z" level=debug msg="Cached ds: datastore1" 
time="2016-07-28T18:33:36Z" level=debug msg="Cached host: " 
time="2016-07-28T18:33:36Z" level=debug msg="Cached pool: " 
time="2016-07-28T18:33:36Z" level=debug msg="vSphere resource cache populated..." 
time="2016-07-28T18:33:36Z" level=debug msg="Looking for scsi controller with label SCSI0" 
time="2016-07-28T18:33:36Z" level=debug msg="Found pvscsi controller directory: 0000:03:00.0" 
time="2016-07-28T18:33:36Z" level=debug msg="Disk location format: /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:%d:0" 
time="2016-07-28T18:33:36Z" level=info msg="Create/attach vmdk [datastore1] 9b52811b-2b1d-47-diskTest/scratch.vmdk from parent " 
time="2016-07-28T18:33:36Z" level=error msg="vmdk storage driver failed to attach disk: Another task is already in progress." 
--- FAIL: TestCreateFS (0.84s)
        Error Trace:    disk_ext4_test.go:74
        Error:          Received unexpected error "Another task is already in progress."
```

This is on a nested esx (4gb) with a devbox vm (2gb) running the storage unit tests.

( `sudo -E VIC_ESX_TEST_URL=https://root:@192.168.8.129/sdk VIC_ESX_TEST_DATASTORE=datastore1 go test -v github.com/vmware/vic/lib/portlayer/storage/vsphere/... github.com/vmware/vic/lib/portlayer/storage/... github.com/vmware/vic/pkg/vsphere/disk/...`)

When printing out the task IDs that we create (by adding a log line in govmomi to `NewTask`), the tasks being reported as conflicting don't match up.  For instance, the test will create and wait on Task A, followed by Task C.  The task number (A or C in this case) is returned by ESX.  In our example, C will return an error (like the one above) and the esx log will report C was in error because task B was in progress.  Our test never created B.  So the issue is "where did B come from"?

**Comment by fdawg4l** [comment]
This issue seems to manifest the most on resource constrained instances.  I have not see it anywhere else.

**Comment by mhagen-vmware** [comment]
In progressing this, as we think that @matthewavery PR may fix this: #2058

**Comment by matthewavery** [comment]
pr #2141 implemented retry logic around TaskInProgress faults which should handle this issue.
